### PR TITLE
Initial work at including I18n in all Slayer objects.

### PR DESCRIPTION
**Pull request by Crisfole**
This mirrors the behavior seen in views in Rails.  Here's the search algorithm:

 1. If the key does not start with a `.`, return `I18n.[METHOD](key, options.dup)`
 2. Otherwise, split the class name by `::`, map each module/class in the hierarchy through `underscore`.
 3. Take the final class name off the hierarchy.
 4. Prepend the class name type ("command" or "service" or "form") to the hierarchy after pluralizing.
 5. Append the rest of the class name to the end of the hierarchy.
 6. Join the hierarchy w/ `.`

Sample for the class `GoodCustomer::Product::UpdateWidgetFactoryCommand.new.t(".key")`

with the following translation file:

```
en:
  commands:
    customer:
      product:
        update_widget_factory:
          key
```

1. Starts with `.` continue:
2. Split results in: `[ "good_customer", "product", "update_widget_factory_command" ]`
3. Class name is `update_widget_factory_command`
4. the Split is now: `[ "commands", "good_customer", "product" ]`
5. The Split is now `[ "commands", "good_customer", "product", "update_widget_factory" ]`
6. Use the key: `commands.good_customer.product.update_widget_factory`

### CODE REVIEW

**Comment by Crisfole** [comment]
Known issues:

 * Does not cache (at class level) the implied parent key for keys with leading dots.
 * Relies on correct naming (`_command.rb` or `_form.rb`). Should instead look at class.

File: Appraisals [hunk 1]
```diff
@@ -4,4 +4,5 @@ end
 
 appraise "rails-5.0" do
   gem "rails", "5.0.0"
+  gem "tzinfo-data"
```
Comment (wkirby) [inline Appraisals:8]: This also needs to be added to the `rails-4.2` section

File: gemfiles/rails_5.0.gemfile [hunk 2]
```diff
@@ -3,5 +3,6 @@
 source "https://rubygems.org"
 
 gem "rails", "5.0.0"
+gem "tzinfo-data"
```
Comment (wkirby) [inline gemfiles/rails_5.0.gemfile:6]: This also needs to be added to the `gemfiles/rails_4.2.gemfile`

**Review by wkirby — CHANGES_REQUESTED** [review]

File: gemfiles/rails_5.0.gemfile (hunk 2 above)
Comment (Crisfole) [reply gemfiles/rails_5.0.gemfile:6]: Still done?

Comment (Crisfole) [reply gemfiles/rails_5.0.gemfile:6]: This file is autogenerated

**Comment by Crisfole** [comment]
@wkirby ready to go?

**Review by wkirby — APPROVED** [review]

**Pull request merged**